File: src/main/java/org/distril/beengine/console/ConsoleSender.java
package org.distril.beengine.console;

import lombok.extern.log4j.Log4j2;
import org.distril.beengine.command.CommandSender;

@Log4j2
public class ConsoleSender implements CommandSender {

	@Override
	public void sendMessage(String message) {
		log.info(message);
	}

	@Override
	public boolean hasPermission(String permission) {
		return true;
	}

	@Override
	public boolean isConsole() {
		return true;
	}

	@Override
	public String getName() {
		return "CONSOLE";
	}
}

File: src/main/java/org/distril/beengine/terminal/ConsoleSender.kt
package org.distril.beengine.terminal

import org.apache.logging.log4j.LogManager
import org.distril.beengine.command.CommandSender

class ConsoleSender : CommandSender {

	override val name: String = "CONSOLE"

	override val isConsole: Boolean = true

	override fun sendMessage(message: String) = log.info(message)

	override fun hasPermission(permission: String) = true

	companion object {

		private val log = LogManager.getLogger(ConsoleSender::class.java)
	}
}
